samza-api/src/main/java/org/apache/samza/system/SystemFactory.java

```diff
@@ -32,4 +32,16 @@
   SystemProducer getProducer(String systemName, Config config, MetricsRegistry registry);
 
   SystemAdmin getAdmin(String systemName, Config config);
+
+  default SystemConsumer getConsumer(String systemName, Config config, MetricsRegistry registry, String consumerIdPrefix) {
```

Comment:
Please add method Javadocs clarifying what the parameters are, and how these methods are different than the ones above.

Reply:
Also, let's make this the second parameter after systemName to group naming related parameters together.

Reply:
good point. Thanks. Will do.

Reply:
> Also, let's make this the second parameter after systemName to group naming related parameters together.

I think change order may not be a good option. We are adding functions to help make previous ones more flexible, so may be better to keep the order the same as previous one. And also, systemName is not intended to be a naming parameter. So i think it may be better to keep it in current order. what do you think？

Reply:
That's a good point, let's keep this order and add javadocs.